### DISCUSSION TITLE
How to validate part of the form

### DISCUSSION THREAD

**Discussion by dstoyanoff** [react-hook-form/react-hook-form#9862] [Q&A]
Hello,
I have a specific need that I can't find a way to achieve using rhf. I have one big form. One of the fields in that form is an array (`useFieldArray`). Every item of that array has 2 sections in the UI. I need a way to know the state of each section (`isValid/isDirty`). The dirty part kind of works by inspecting `formState.dirtyFields`, but `formState.errors` is an empty object and I can't find a way to get the errors before all fields were dirtied. 

I cannot use different forms due to the parent level array and also because the whole thing is submitted at once.

Here is the yup schema (not the real one, but explains the point):
```
yup.object({
  items: yup.array().of(yup.object({
    section1: yup.object({
      field1: yup.string().required(),
      field2: yup.string().optional(),
    }),
    section2: yup.object({
      field3: yup.string().required(),
      field4: yup.string().optional(),
    })
  }))
})
```

From the above example, I need to be able to know if `items.0.section1` is valid or not, which I can't currently find a way to do

**Comment by Moshyfawn** [comment]
Have you seen [the `trigger` method](https://react-hook-form.com/api/useform/trigger)?

Combining it with `useForm` `mode` to set the "base" validation mode, you can call [the `trigger` method](https://react-hook-form.com/api/useform/trigger) whenever you need to in depending use-case

**Reply by dstoyanoff** [reply]
Trigger is not the best option as it's only ran on demand. I need a reactive value so everything is recalculated while you edit the fields. 

For now I've settled using yup directly
